src/shared/ui/survey/AudioRecorderItem.tsx

```diff
@@ -0,0 +1,195 @@
+import { useState, useRef, useEffect, useCallback, useMemo, FC } from 'react';
+import React from 'react';
+import { TouchableOpacity } from 'react-native';
+import { Alert } from 'react-native';
+
+import { useFocusEffect } from '@react-navigation/native';
+import { useTranslation } from 'react-i18next';
+import AudioRecorderPlayer, {
+  AudioSet,
+} from 'react-native-audio-recorder-player';
+import Permissions, { RESULTS, openSettings } from 'react-native-permissions';
+import RNFetchBlob from 'rn-fetch-blob';
+
+import {
+  getMicrophonePermissions,
+  IS_ANDROID,
+  randomString,
+} from '@app/shared/lib';
+import { useMicrophonePermissions } from '@shared/lib';
+import { StopIcon, MicrophoneIcon, XStack, Text, YStack } from '@shared/ui';
+
+const audioSetConfig: AudioSet = {
+  AVNumberOfChannelsKeyIOS: 1,
+};
+const androidCacheDir = RNFetchBlob.fs.dirs.CacheDir;
+
+type Props = {
+  config: {
+    maxLength?: number;
+  };
+  onChange: (path: string) => void;
+};
+
+const AudioRecorderItem: FC<Props> = ({ config, onChange: onFinish }) => {
+  const audioRecorderPlayer = useRef(new AudioRecorderPlayer());
+  const { t } = useTranslation();
+  const { maxLength = Infinity } = config;
+  const microphonePermission = useMicrophonePermissions();
+  const [isRecording, setIsRecording] = useState(false);
+  const [secondsElapsed, setSecondsElapsed] = useState(0);
+  const [fileSaved, setFileSaved] = useState(false);
+  const [errorDescription, setErrorDescription] = useState('');
+  const permissionNotGranted = microphonePermission !== RESULTS.GRANTED;
+
+  const filePath = useMemo(
+    () =>
+      IS_ANDROID
+        ? `${androidCacheDir}/${randomString(20)}.mp4`
+        : `${randomString(20)}.m4a`,
+    [],
+  );
+
+  const checkMicrophonePermission = async () => {
+    if (permissionNotGranted) {
+      const result = await getMicrophonePermissions();
+      if (result === Permissions.RESULTS.BLOCKED) {
+        return await handleBlockedPermissions();
+      }
+    }
+    return true;
+  };
+
+  const handleBlockedPermissions = async () =>
```

Comment:
Please move the function to the `lib` segment.
You can create a folder: shared/lib/alerts